test/integration/resourcemanager/managedresource/reconciler_test.go

```diff
@@ -0,0 +1,213 @@
+				return testClient.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)
+			}, time.Minute, 5*time.Second).Should(BeNotFoundError())
+			Eventually(func() error {
+				return testClient.Get(ctx, client.ObjectKeyFromObject(configMap), configMap)
+			}, time.Minute, 5*time.Second).Should(BeNotFoundError())
+			Expect(testClient.Delete(ctx, secretForRef)).To(Or(Succeed(), BeNoMatchError()))
+		})
+
+		Context("create managed resource", func() {
+			It("should successfully create the resources and maintain proper status conditions", func() {
+				err = testClient.Create(ctx, managedResource)
+				Expect(err).To(Or(Succeed(), BeAlreadyExistsError()))
+
+				Eventually(func() error {
+					return testClient.Get(ctx, client.ObjectKeyFromObject(configMap), configMap)
+				}, time.Minute, time.Second).Should(Succeed())
+
+				Eventually(func() bool {
+					err := testClient.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)
+					condition := gardenerv1beta1helper.GetCondition(managedResource.Status.Conditions, resourcesv1alpha1.ResourcesApplied)
+					return err == nil && condition != nil && condition.Status == gardencorev1beta1.ConditionTrue
+				}, time.Minute, time.Second).Should(BeTrue())
+			})
+
+			It("should not be created successfully due to missing secret reference", func() {
+				managedResource.Spec.SecretRefs = []corev1.LocalObjectReference{{Name: "missing"}}
+				err = testClient.Create(ctx, managedResource)
+				Expect(err).To(Or(Succeed(), BeAlreadyExistsError()))
+				Eventually(func() bool {
+					err := testClient.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)
+					condition := gardenerv1beta1helper.GetCondition(managedResource.Status.Conditions, resourcesv1alpha1.ResourcesApplied)
+					return err == nil && condition != nil && condition.Status == gardencorev1beta1.ConditionFalse && condition.Reason == "CannotReadSecret"
+				}, 30*time.Second, time.Second).Should(BeTrue())
+			})
+
+			It("resource creation should fail due to incorrect object", func() {
+				configMap := &corev1.ConfigMap{}
+				data, err := createSecretDataFromObject(configMap, "config-map.yaml")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(data).ToNot(BeNil())
+				secretForRef.Data = *data
+				Expect(testClient.Update(ctx, secretForRef)).To(Succeed())
+				err = testClient.Create(ctx, managedResource)
+				Expect(err).To(Or(Succeed(), BeAlreadyExistsError()))
+				Eventually(func() bool {
+					err := testClient.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)
+					condition := gardenerv1beta1helper.GetCondition(managedResource.Status.Conditions, resourcesv1alpha1.ResourcesApplied)
+					return err == nil && condition != nil && condition.Status == gardencorev1beta1.ConditionFalse
+				}, 30*time.Second, time.Second).Should(BeTrue())
+			})
+		})
+
+		Context("update managed resource", func() {
+			It("the managed resource should be updated successfully", func() {
```

Comment:
```suggestion
			It("should successfully create new resources by adding new resources to the secret", func() {
```

I may be wrong about my suggestion but I would expect to see what kind of update exactly we are testing with this test case. I am struggling to follow. Can you explain the scenario that is tested?

Reply:
@ialidzhikov I think the scenario here is to check whether the managed resource gets reconciled if one of the secret that it refers to is updated: https://github.com/gardener/gardener/blob/f3e414ecd8ff2b0a2143e569b57f93a25a18dc9f/pkg/resourcemanager/controller/managedresource/add.go#L113-L121
I'll adapt the description appropriately.

I just don't understand the `BeNoMatchError()` call in this test case though: https://github.com/gardener/gardener/blob/3a2137e852113fa76d64d76acbbb88b102008490/test/integration/resourcemanager/managedresource/reconciler_test.go#L152